### ISSUE TITLE
Remove support for single color transparency

### ISSUE THREAD

**Issue by cmb69** [libgd/libgd#373]
libgd still supports single color transparency by means of [`gdImage.transparent`](https://github.com/libgd/libgd/blob/gd-2.2.4/src/gd.h#L393). This was done for backward compatibility with GD 1, which didn't support full alpha channels, and doesn't appear to be necessary nowadays (semi-transparency is [supported by "all" somewhat contemporary browsers](http://caniuse.com/#feat=png-alpha)). The benefit of removing the support for single color transparency would be simplification of the code base, and also easier understanding for our users. For instance, [`gdImageCopyMerge()`](http://libgd.github.io/manuals/2.2.4/files/gd-c.html#gdImageCopyMerge) ignores alpha channels, but respects the single color transparency, what obviously [causes some confusion and different behavior](https://bugs.php.net/bug.php?id=50183).

**Comment by cmb69** [comment]
Hm, not sure if that would work for GIFs, though.

**Comment by pierrejoye** [comment]
I agree, we should drop it at the image level. We can't without BC breaks.

gdSurface has no such thing, that's a beginning.

However it may be useful at the codecs level, where it should have been done in the first place anyway.
